Lesson6

Hanna-Konovalova opened the pull request:
lesson6 Homework completed

Hanna-Konovalova committed:
lesson4: Homework completed

Hanna-Konovalova committed:
lesson5: Homework completed

Hanna-Konovalova committed:
lesson5: Task1 remarks corrected

Hanna-Konovalova committed:
lesson5: Task1 remarks corrected

Hanna-Konovalova committed:
lesson5: Task1 remarks corrected 20220620

Hanna-Konovalova committed:
lesson5: Homework completed

RomanKashenok commented on Part1/Part1.iml:
```diff
@@ -0,0 +1,11 @@
+    <orderEntry type="inheritedJdk" />
+    <orderEntry type="sourceFolder" forTests="false" />
+  </component>
+</module>
```
надо убрать это (добавить в гитигнор)

RomanKashenok commented on Part1/src/com/konovalova/lesson4/HomeworkTask2.java:
```diff
@@ -0,0 +1,25 @@
+
+        System.out.println("Number1 >= Number2 " + res);
+
+      in.close();
```
вот тут - очень найс ))

RomanKashenok commented on Part1/src/com/konovalova/lesson4/HomeworkTask3.java:
```diff
@@ -0,0 +1,21 @@
+        Scanner in = new Scanner(System.in);
+        System.out.print("Input a four-digit number: ");
+        String str = in.nextLine();
+        String reverse = new StringBuffer(str).reverse().toString();
```
это не число - ты просто вывела строку в обратном порядке. Надо получить число. Если б я просил перевернуть строку - ок. Но тут речь конкретно о 4х значном числе

RomanKashenok commented on Part1/src/com/konovalova/lesson4/HomeworkTask4.java:
```diff
@@ -0,0 +1,28 @@
+        int bb = (aa - b*100);
+        int c = bb/10;
+        int cc = (bb - c*10);
+        boolean res = (a == b | b == c | c == cc | a == c | a == cc | b == cc);
```
зачем нам проверять все варианты, если, возможно a == b -> true?

RomanKashenok commented on Part1/src/com/konovalova/lesson5/HomeworkTask1.java:
```diff
@@ -0,0 +1,26 @@
+        System.out.println(s4);
+        //5
+        String s5 = new StringBuilder().append(a).toString();
+        System.out.println(s5);
```
отлично!

RomanKashenok commented on Part1/src/com/konovalova/lesson5/HomeworkTask2.java:
```diff
@@ -0,0 +1,106 @@
+        /* Написать программу, в которой в консоль будет выводится название месяца и сезон,
+        * к которму этот месяц относится (написать с помощью  if  и с помощью switch)
+        **/
+        Scanner in = new Scanner(System.in);
```
давай сделаем это всё в одном switch-case :) и пору года выведем и название месяца

RomanKashenok commented on Part1/src/com/konovalova/lesson6/HomeworkTask1.java:
```diff
@@ -0,0 +1,30 @@
+
+        double averageMult = 0;
+        double Mult = 1;
+        int SymbolQuant = 0;
```
имена переменных с маленькой буквы

RomanKashenok commented on Part1/src/com/konovalova/lesson6/HomeworkTask2.java:
```diff
@@ -0,0 +1,40 @@
+        if (IsThreeDigit) {
+
+            System.out.println("Number is 3 digital");
+            int lastNumeric = ourNumeric - ((ourNumeric/10)*10);
```
не порще ли % 100 == 7 ?

RomanKashenok commented on Part1/src/com/konovalova/lesson6/HomeworkTask4.java:
```diff
@@ -0,0 +1,25 @@
+         } else if (lastNumeric>1 && lastNumeric<5) {
+            System.out.println(randNum + " рубля");
+        } else {
+            System.out.println(randNum + " рублей");
```
а если 21?

RomanKashenok commented on Part1/src/com/konovalova/lesson6/HomeworkTask4.java:
```diff
@@ -0,0 +1,25 @@
+        if(randNum == 1) {
+            System.out.println(randNum + " рубль");
+         } else if (lastNumeric>1 && lastNumeric<5) {
+            System.out.println(randNum + " рубля");
```
а если 12, 13?

RomanKashenok commented on Part1/src/com/konovalova/lesson6/HomeworkTask5.java:
```diff
@@ -0,0 +1,35 @@
+
+        boolean isRemain = true;
+
+        while (isRemain) {
```
давай это всё оптимизируем. 
Первый результат от деления - а/а == 1 (это все числа умеют), следующий а/? == 2. Соответстенно надо проверить не все числа от нуля до "а", а числа от 2 до а/2, верно? Если еще дальше оптимизировать, то подумай на счет корня квадратного из числа )

RomanKashenok commented on Part1/src/com/konovalova/lesson6/HomeworkTask6.java:
```diff
@@ -0,0 +1,27 @@
+    public static void main(String[] args) {
+        /* Найдите сумму первых n целых чисел, которые делятся на 3.
+        * */
+        Scanner in = new Scanner(System.in);
```
не верно. Если я хочу найти сумму первых 5ти чисел, которые делятся на 3, то это будет 3+6+9+12+15 == 45, а не 27 как в твоем случае

RomanKashenok commented on Part1/src/com/konovalova/lesson6/HomeworkTask7.java:
```diff
@@ -0,0 +1,22 @@
+public class HomeworkTask7 {
+    public static void main(String[] args) {
+        /* Посчитать факториал числа в границах от 10 до 15 (используя рекурсию)
+         * */
```
факториал числа в границах от 10 до 15 - это факториал числа 13, например. А не то, что у тебя ниже

RomanKashenok commented on Part1/src/com/konovalova/lesson6/HomeworkTask8.java:
```diff
@@ -0,0 +1,18 @@
+public class HomeworkTask8 {
+    public static void main(String[] args) {
+        /* Посчитать факториал числа в границах от 10 до 15 (не используя рекурсию, в цикле)
+        * */
```
аналогично предыдущему

RomanKashenok commented on Part1/src/com/konovalova/lesson6/HomeworkTask9.java:
```diff
@@ -0,0 +1,59 @@
+            } else if (compereResult > 10) {
+                System.out.println("Cold");
+            }
+        }
```
почти гуд )) не все варианты в выводе получают подсказку (если загадано y (как у меня), то после ввода х - просто пустота)

Hanna-Konovalova committed:
lesson 3 HomeworkTask3 program algorithm changed
lesson 4 HomeworkTask4 fixed comparison conditions
lesson 5 HomeworkTask2 program algorithm changed from if/else & switch/case to switch/case
lesson 6 HomeworkTask2 logic of searching last number 7 is changed
lesson 6 HomeworkTask4 program algorithm changed, paid attention to numbers 21,13,14 etc.
lesson 6 HomeworkTask9 added condition for 1 symbol difference
lesson 6 HomeworkTask6 program algorithm changed
lesson 6 HomeworkTask7-8 changed factorial principe account
lesson 6 HomeworkTask5 program logic changed with using of sqrt method

Hanna-Konovalova committed:
lesson 3 HomeworkTask3 program algorithm changed
lesson 4 HomeworkTask4 fixed comparison conditions
lesson 5 HomeworkTask2 program algorithm changed from if/else & switch/case to switch/case
lesson 6 HomeworkTask2 logic of searching last number 7 is changed
lesson 6 HomeworkTask4 program algorithm changed, paid attention to numbers 21,13,14 etc.
lesson 6 HomeworkTask9 added condition for 1 symbol difference
lesson 6 HomeworkTask6 program algorithm changed
lesson 6 HomeworkTask7-8 changed factorial principe account
lesson 6 HomeworkTask5 program logic changed with using of sqrt method

RomanKashenok commented on Part1/src/com/konovalova/lesson6/HomeworkTask6.java:
```diff
@@ -8,10 +8,11 @@ public static void main(String[] args) {
-        while(counter!= 3) {
+        while(counter!= counterLimit) {
 
             boolean isDivided3 = firstNum%3 == 0;
```
Ок, но если брать во внимание, что мы ищем числа, которые делятся на 3, то зачем нам рассматривать 4,5,7.8 и т.д. Мы же можем просто сложить 3 + 6+ 9 ... т.е. делать не firstNum++ , а firstNum+=3 и тогда лишняя проверка не нужна

RomanKashenok commented on Part1/src/com/konovalova/lesson6/HomeworkTask8.java:
```diff
@@ -1,15 +1,19 @@
+        int i=1;
 
-        for(i=firstNumber;i<=lastNumber;i++) {
+        for(;i<=randNum;i++) {
```
объявляй i в условии цикла - так все же более читабельно

RomanKashenok reviewed (approved):
мелкие комментарии

Hanna-Konovalova committed:
lesson 6 HomeworkTask6 - change counter condition
lesson 6 HomeworkTask8 - add variable initialization in the loop
